lib.sh

```diff
@@ -79,7 +79,7 @@ _commit_if_needed() {
 }
 
 _lint_and_fix_action() {
-	if [[ $GITHUB_EVENT_NAME == "push" ]]; then
+	if [[ $GITHUB_EVENT_NAME == "push" || $GITHUB_EVENT_NAME == "pull_request" ]]; then
```

Comment:
From what I've understood from [the documentation](https://developer.github.com/v3/activity/events/types/#pullrequestevent), this event will be triggered when new events modify the Pull Request itself, not when code changes on the PR.

Was there a bigger reason why this was required? If that is the case, I think we need to consider only the relevant `action` argument, and avoid triggering the action when a PR is closed.

The only one I could imagine so far is if `push` is not being emitted when a PR is proposed by a fork from another contributor. But also according to the [documentation](https://developer.github.com/actions/managing-workflows/workflow-configuration-options/#pull-request-events-for-forked-repositories), it's better to base only on `push`:

> If you intend to use the pull_request event to trigger CI tests, it's recommended to listen to the push event.

Reply:
Oh, wow, thanks for the enlightment. 

The intention is/was to only trigger the prettier action when a PR is created (not draft) or new code is pushed.

Looks like i have to review the trigger in my test cases, probably  i will  revert the change then.

---

On a side thought: Do you think the `$GITHUB_EVENT_NAME` on which the action trigger should be the concern of the library instead of the consumer?

Reply:
Draft PRs is something quite new, so I'm not used on how the payload information arrives.
The use case for avoid filtering the actions on Draft PRs would be a great addition to the [actions/bin/filter](https://github.com/actions/bin/tree/master/filter) IMO.

Generally, from what I've seen, most actions react only to a single event. The same actions on this repo could act differently on different scenarios, that is why it considers the `$GITHUB_EVENT_NAME`. Given the payload for a `push` and a `review` are different, it needs to know the context it has been triggered.

Reply:
We would like to automatically fix-up code when a branch is opened or a change is pushed to a branch with an open PR.

After investigating, this is only possible with the `PULL_REQUEST` event. Thus i would like to keep it and add the additional action check.

Is that ok with you @bltavares?

Reply:
I don't think it is necessary to include. From my experience, listening to `push` is enough for what you intend. I've been running this exact same goal, and using only `push`. This repo has several examples, notice the `shfmt` action:

- *Fix directly to push on master* https://github.com/bltavares/actions/commit/115e3da
- *PR with multiple fixes* [1](https://github.com/bltavares/actions/pull/8) [2](https://github.com/bltavares/actions/pull/10)

I've been using [this workflow](https://gist.github.com/bltavares/d2eb4b9bca7869b0bd12e36337859885) which fixes internal PRs as soon as a new PR is opened, and when new commits are pushed on several repos. 

Would you mind sharing your workflow so I can better understand your setup?

Reply:
You could even try it here. Mess up the indentation of any `.sh` and the action should fix it again.

Reply:
Wouldn't the suggested worklow also trigger when something is pushed to the master branch?

And reacting on every push would quickly reach the limit of github actions (two parallel) which could become a bottleneck, even when adding a branch filter.

Our current workflow is rather simple and looks as follows (php-cs-fixer is a linter we also plan to contribute).

    workflow "apply linters on pull request" {
      on = "pull_request"
      resolves = ["prettier", "php-cs-fixer"]
    }

    action "prettier" {
      uses = "rebuy-de/actions/prettier@prettier"
      args = ["autofix"]
      secrets = ["GITHUB_TOKEN"]
      needs = ["action-filter"]
    }

    action "php-cs-fixer" {
      uses = "rebuy-de/actions/php-cs-fixer@php-cs-fixer"
      args = ["autofix"]
      secrets = ["GITHUB_TOKEN"]
      needs = ["action-filter"]
    }

    action "action-filter" {
      uses = "actions/bin/filter@master"
      args = "action 'opened|ready_for_review|synchronize'"
    }

Reply:
Maybe it is better to use an environment variable which defines the events on which an `autofix` should happen. 

Example: 

```workflow
action "shfmt" {
  uses = "./shfmt"
  args = ["autofix"]
  env = {
    AUTOFIX_EVENTS="pull_request,push"
  }
  secrets = ["GITHUB_TOKEN"]
}
```

Then everyone can simply define the events they are interested in for autofix while the default will always be the `push` event (to keep backward compatibility). 

What do you think?

Reply:
Hey, i implemented @spea's suggestion which will enable the user of the lib to define the actions as env variable :)